libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/util/MapboxDropInUtils.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.dropin.util
+
+import android.annotation.SuppressLint
+import android.content.Context
+import com.mapbox.android.core.location.LocationEngineCallback
+import com.mapbox.android.core.location.LocationEngineProvider
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+
+internal object MapboxDropInUtils {
+
+    @SuppressLint("MissingPermission")
+    fun getLastLocation(
+        context: Context,
+        resultConsumer: (Expected<Exception, LocationEngineResult>) -> Unit
+    ) {
+        LocationEngineProvider.getBestLocationEngine(context.applicationContext).getLastLocation(
```

Comment:
We might need to keep a weak reference to the `resultConsumer` here otherwise we might leak. I remember location engine being able to leak in the past since there's no way to remove the queued update request.